frontend/projects/upgrade/src/app/features/dashboard/experiment-users/components/stratification-factor-list/stratification-factor-list.component.ts

```diff
@@ -0,0 +1,90 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ImportStratificationsComponent } from './import-stratifications/import-stratifications.component';
+import { MatLegacyDialog as MatDialog } from '@angular/material/legacy-dialog';
+import * as clonedeep from 'lodash.clonedeep';
+import { DeleteStratificationComponent } from './delete-stratification/delete-stratification.component';
+import { StratificationFactor } from '../../../../../core/stratification-factors/store/stratification-factors.model';
+import { Subscription } from 'rxjs';
+import { StratificationFactorsService } from '../../../../../core/stratification-factors/stratification-factors.service';
+
+@Component({
+  selector: 'users-stratification-factor-list',
+  templateUrl: './stratification-factor-list.component.html',
+  styleUrls: ['./stratification-factor-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class StratificationComponent implements OnInit {
+  allStratificationFactors: StratificationFactor[];
+  allStratificationFactorsSub: Subscription;
+  isLoadingStratificationFactors$ = this.stratificationFactorsService.isLoadingStratificationFactors$;
+  stratificationFactorsForTable: { factor: string; summary: string }[] = [];
+  displayedColumns: string[] = ['factor', 'summary', 'actions'];
+
+  constructor(private dialog: MatDialog, private stratificationFactorsService: StratificationFactorsService) {}
+
+  ngOnInit(): void {
+    this.allStratificationFactorsSub = this.stratificationFactorsService.allStratificationFactors$.subscribe(
+      (allStratificationFactors) => {
+        this.allStratificationFactors = allStratificationFactors;
+        this.stratificationFactorsForTable = this.convertToTableformat();
+      }
+    );
+  }
+
+  convertToTableformat() {
+    const stratificationFactors = this.allStratificationFactors.map((element) => {
```

Comment:
instead of "element", can we say "stratificationFactor" so it's obvious?

I'd like to see a type defined for this "table format", it has an implied return type of 

```
{
    factor: string;
    summary: string;
}[]
```

Which feel like an important thing to define an explicit type for. Maybe something like:

```
interface StratificationFactorsTableRow {
    factor: string;
    summary: string;
}

type StratificationFactorsTable = StratificationFactorTableRow[];
```